### PR TITLE
🌱 Add exp/util MachinePool helpers

### LINKED ISSUE #3391: Move CAPZ machinePool helper functions into CAPI's util package
**User Story**

As we dive into the AWSMachinePool implementation in CAPA, we're finding some nice things in CAPZ that we want to shamelessly steal, most notably the `getMachinePoolByName` and `getOwnerMachinePool` helpers. Any MachinePool provider will need to use this functionality. 

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6ad50d0e79546d9f3c5e5d632836412df91a6a9c/exp/controllers/azuremachinepool_controller.go#L598-L622

/kind feature

### PR DESCRIPTION
**Pull request by rudoi**
**What this PR does / why we need it**:

Adds some helpers that all MachinePool providers will need. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3391

### CODE REVIEW

**Comment by detiber** [comment]
@rudoi I'm wondering if it would make sense to have a `util` package under the `exp` directory for utilities that are only used for experimental features. It would help avoid exposing these types of functions out as part of the api that is more generally consumed and would allow us to have a bit more leeway in making breaking changes as needed for experimental things without it affecting the "stable" api.

**Comment by rudoi** [comment]
@detiber yep, makes sense to me. will move.

File: exp/util/util.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package util
```
Comment (detiber) [inline exp/util/util.go:1]: missing license and copyright boilerplate here.

Comment (rudoi) [reply exp/util/util.go:17]: caught myself - need to get into the habit of `make verify` before pushing 🤦

**Pull request merged**